scripts/justmat/paths.lua

```diff
@@ -203,36 +256,42 @@ function init()
   
   black_keys = keybed_leds()
 
-  if playing then
-    clk:start()
-  else clk:stop() end
+  clk:start()
 
   params:read("justmat/paths.pset")
   params:bang()
 end
 
 
 function step()
-  if current_step == seq_data[current_seq].steps and end_of_seq_change then
-    current_seq = next_seq
-    end_of_seq_change = false
+  if paths[current_path].current_step == 1 and paths[current_path].end_of_pat_change then
```

Comment:
Oh! This is nice. I have been fretting over how long some of my lines are, and this would shorten quite a few. I am still fairly new to coding, so I eat this sort of critique up. :)

Reply:
well, your code is great and do keep in mind my opinions are entirely just that.  in general i do think that repeated bits make for code that's harder to read and maintain.  DRY (don't repeat yourself) ([reference](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself)) is a kind of engineering mantra and i generally strive for it though taken too far it's no fun.  anyway, yeah.  always happy to look over your stuff! 👍

Reply:
I ended up leaving this one as is.